src/pkg/cli/permissions/permissions_test.go

```diff
@@ -0,0 +1,91 @@
+package permissions
+
+import (
+	"strings"
+	"testing"
+
+	defangv1 "github.com/DefangLabs/defang/src/protos/io/defang/v1"
+)
+
+func TestPermission(t *testing.T) {
+	test := []struct {
+		name              string
+		action            ActionRequest
+		expectedResult    bool
+		expectedErrorText string
+	}{
+		{
+			name: "no permission for gpu",
+			action: ActionRequest{
+				role:     defangv1.SubscriptionTier_PERSONAL,
+				action:   "compose-up",
+				resource: "gpu",
+			},
+			expectedErrorText: "gpus not supported by this tier",
+		},
+		{
+			name: "have permission for gpu",
+			action: ActionRequest{
+				role:     defangv1.SubscriptionTier_PRO,
+				action:   "compose-up",
+				resource: "gpu",
+				detail:   "1",
```

Comment:
We already have a quota structure, so kinda expected us to leverage, extend, or replace that one.